src/plugins/telemetry_server/telemetry_server_impl.h

```diff
@@ -62,14 +68,21 @@ class TelemetryServerImpl : public PluginImplBase {
         bool gyro_status,
         bool accel_status,
         bool mag_status,
-        bool gps_status) const;
+        bool gps_status);
 
     TelemetryServer::Result publish_extended_sys_state(
-        TelemetryServer::VtolState vtol_state, TelemetryServer::LandedState landed_state) const;
+        TelemetryServer::VtolState vtol_state, TelemetryServer::LandedState landed_state);
 
 private:
     std::chrono::time_point<std::chrono::steady_clock> _start_time;
 
+    std::vector<RequestMsgInterval> _interval_requests;
+    std::mutex _interval_mutex;
+    std::mutex _msg_cache_mutex;
```

Comment:
Is `_msg_cache_mutex` really required? It's only used once where we also have `_interval_mutex`.

Reply:
Yes I was having the same thought. I suppose I just want to make sure each data store is locking on access :)